lib/core/src/display/world.rs

```diff
@@ -28,221 +42,262 @@ use eval_tt::*;
-            Some(item) => group!(logger, "Dropping workspace {}.", id, {
-                let _destruct_it_here = item;
-            }),
-        }
-    }
-
-    /// Run the provided callback on every frame. Returns a `CallbackHandle`,
-    /// which when dropped will cancel the callback. If you want the function
-    /// to run forever, you can use the `forget` method in the handle.
-    pub fn on_frame<F:FnMut(&mut World)+'static>
-    (&mut self, mut callback:F) -> CallbackHandle {
-        let this = self.self_reference.as_ref().unwrap().clone_rc();
-        let func = move || callback(&mut this.borrow_mut());
-        self.event_loop.add_callback(func)
+    fn new_uninitialized<Dom:Str>(dom:Dom) -> Self {
+        let stats                  = default();
+        let logger                 = Logger::new("world");
+        let workspace_logger       = logger.sub("workspace");
+        let workspace_dirty_logger = logger.sub("workspace_dirty");
+        let workspace_dirty        = WorkspaceDirty::new(workspace_dirty_logger,());
+        let on_change              = workspace_on_change(workspace_dirty.clone_ref());
+        let variables              = UniformScope::new(logger.sub("global_variables"));
+        let time                   = variables.add_or_panic("time",0.0);
+        let display_mode           = variables.add_or_panic("display_mode",0);
+        let workspace              = Workspace::new(dom,&variables,workspace_logger,&stats,on_change).unwrap(); // fixme unwrap
+        let fonts                  = Fonts::new();
+        let event_loop             = EventLoop::new();
+        let update_handle          = default();
+        let stats_monitor          = StatsMonitor::new(&stats);
+        let performance            = web::get_performance().unwrap();
+        let start_time             = performance.now() as f32;
+        let stats_monitor_cp_1     = stats_monitor.clone();
+        let stats_monitor_cp_2     = stats_monitor.clone();
+        event_loop.set_on_loop_started  (move || { stats_monitor_cp_1.begin(); });
+        event_loop.set_on_loop_finished (move || { stats_monitor_cp_2.end();   });
+        Self {workspace,workspace_dirty,logger,event_loop,variables,performance,start_time,time,display_mode,fonts,update_handle,stats,stats_monitor}
     }
 
     pub fn run(&mut self) {
+        let relative_time = self.performance.now() as f32 - self.start_time;
+        self.time.set(relative_time);
         self.update();
     }
 
     /// Check dirty flags and update the state accordingly.
     pub fn update(&mut self) {
-        if self.workspace_dirty.check_all() {
-            group!(self.logger, "Updating.", {
-                // FIXME render only needed workspaces.
-                self.workspace_dirty.unset_all();
-                let fonts = &mut self.fonts;
-                self.workspaces.iter_mut().for_each(|t| t.update(fonts));
+        //        if self.workspace_dirty.check_all() {
```

Comment:
What is the reason of commenting this code? Uncomment, remove, or add "TODO" comment